sdk/containerregistry/Azure.Containers.ContainerRegistry/src/Blob/BlobHelper.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Security.Cryptography;
+using System.Text;
+
+namespace Azure.Containers.ContainerRegistry.Specialized
+{
+    internal class BlobHelper
+    {
+        internal static string ComputeDigest(Stream stream)
+        {
+            // According to https://docs.docker.com/registry/spec/api/#content-digests, compliant
+            // registry implementations use sha256.
+
+            using SHA256 sha256 = SHA256.Create();
+            var position = stream.Position;
+
+            try
+            {
+                stream.Position = 0;
+                var hashValue = sha256.ComputeHash(stream);
+                return FormatDigest(hashValue);
+            }
+            finally
+            {
+                stream.Position = position;
+            }
+        }
+
+        internal static string FormatDigest(byte[] hash)
+        {
+            return $"sha256:{BytesToString(hash)}";
```

Comment:
would be a bit more efficient to prepend `sha256:` inside `BytesToString` in string builder